docs/data/material/guides/right-to-left/RTLDemo.tsx

```diff
@@ -7,21 +7,20 @@ import { CacheProvider } from '@emotion/react';
 import createCache from '@emotion/cache';
 
 const theme = createTheme({
-  direction: 'rtl', // Both here and <body dir="rtl">
+  direction: 'rtl',
 });
-// Create rtl cache
+
 const cacheRtl = createCache({
   key: 'muirtl',
   stylisPlugins: [prefixer, rtlPlugin],
 });
 
-export default function Direction() {
+export default function RTLDemo() {
```

Comment:
@MBilalShafi spotted in https://github.com/mui/mui-x/issues/9552#issuecomment-1746165289 that this demo has a bug (same in Joy UI)

I think we can fix it as he did for MUI X: https://github.com/mui/mui-x/pull/10728/files#diff-2df740af28be70d2dc467f91142c426f909fbaf665ddf5400b3eb24937cc1668R54

Reply:
Just to understand it correctly: the fix is to use this `const existingTheme = useTheme();` const, which would fix the referencing to the docs theme? This wouldn't be a problem for users actually using the library, right?

I'm wondering if there's away to make the demo work without polluting with docs-specific treatment? Maybe not, that's ok, but just trying to understand and make the demo as simple as possible!